exporter.py

```diff
@@ -0,0 +1,228 @@
+
+#  -------------------------------------------------------------------------
+#  Copyright (C) 2019 Daniel Werner Lima Souza de Almeida -
+#                     dwlsalmeida at gmail dot com
+#  -------------------------------------------------------------------------
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+import bpy
+from . import RamsesPython
+import logging
+from .intermediary_representation import *
+from typing import List
+
+log = logging.getLogger(name='ramses-scene-exporter')
+
+class RamsesBlenderExporter():
+    """Extracts the scene graph, translating it to a RAMSES scene"""
+
+    def __init__(self, scenes: List[bpy.types.Scene]):
+        self.scenes = scenes
+        self.scene_representations = []
+        self.ready_to_translate = False
+
+    def extract_from_blender_scene(self):
+        """Extract the scene graph from Blender, building an internal
+        representation that can then be used to build a RAMSES scene"""
+
+        for scene in self.scenes:
+            extractor = BlenderRamsesExtractor(scene)
+            representation = extractor.run()
+            self.scene_representations.append(representation)
+
+        self.ready_to_translate = True
+
+    def build_ramses_scene(self):
+        """Builds a RAMSES scene out of the available scene \
+            representations
+
+        Raises:
+            RuntimeError: Raised when 'extract_from_blender_scene' is \
+                not called first.
+        """
+
+        if not self.ready_to_translate:
+            raise RuntimeError("Extract data from Blender first.")
+
+        ramses = RamsesPython.Ramses("test")
+
+        ramses_scene = ramses.createScene("test scene")
+
+        ramses_render_group = ramses_scene.createRenderGroup('rendergroup')
+        ramses_render_pass = ramses_scene.createRenderPass('renderpass')
+
+        ramses_render_pass.addRenderGroup(ramses_render_group, 0)
+
+        for scene_representation in self.scene_representations:
+            ir_root = scene_representation.graph.root
+            ramses_root = self.__ramses_build_subscene(ramses_scene, ir_root, parent=None)
+
+            log.debug(f'Successfully built RAMSES Scenegraph: {str(ramses_root)}')
+
+
+        sceneFile = "/tmp/scene.ramses"
+        sceneResources = "/tmp/scene.ramres"
+        ramses_scene.saveToFiles(sceneFile, sceneResources, True)
+        validationReport = str(ramses_scene.getValidationReport())
+
+
+        log.debug(f"Validation report for scene {str(ramses_scene)}:\n{validationReport}")
+        if validationReport != '':
+            raise RuntimeError(validationReport)
+
+
+    def __ramses_build_subscene(self,
```

Comment:
did you intend to use "_ _" demangling prefix here? What for? Do you expect to override the exporter class?

Reply:
No I don't expect to override it. I thought this was a convention meaning 'private' in Python ?

_Sent from my Redmi Note 5 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

Reply:
Not really ;) Read this, it's quite good: https://dbader.org/blog/meaning-of-underscores-in-python

Reply:
But I guess it's the best closest thing... So I guess it's ok to leave the "__". Was just curious :)

Reply:
Oh wow, this is totally not what I was looking for! Looking at that article maybe I was shooting for a single underscore at the beginning then.

_Sent from my Redmi Note 5 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

Reply:
It's interesting to see how python works :) Just choose the one you like better, but stick to it then ;)